### PR TITLE
Incorporate checksum_resources into fetch for git tasks

### PR DESCRIPTION
**Pull request by tomjw64**

### CODE REVIEW

File: bakery/src/tasks/pdfify-single.js [hunk 1]
```diff
@@ -33,13 +32,11 @@ const task = (taskArgs) => {
         { name: fetchedInput },
         { name: fetchedResourcesInput },
         { name: styleInput },
-        { name: symlinkInput },
         { name: mathifiedInput }
       ],
       outputs: [{ name: artifactsOutput }],
       params: {
         ARTIFACTS_OUTPUT: artifactsOutput,
-        SYMLINK_INPUT: symlinkInput,
```
Comment (rnathuji) [inline bakery/src/tasks/pdfify-single.js:42]: I believe the PR is missing the corresponding update to `pdfify_single.sh` where symlinks are injected.

File: bakery/src/tasks/fetch-book-group.js [hunk 2]
```diff
@@ -27,15 +28,17 @@ const task = (taskArgs) => {
...
+        { name: unusedResourceOutput }
       ],
       params: {
         COLUMNS: 80,
         BOOK_INPUT: bookInput,
         GH_SECRET_CREDS: githubSecretCreds,
         CONTENT_OUTPUT: contentOutput,
         BOOK_SLUGS_URL: bookSlugsUrl,
-        RESOURCE_OUTPUT: resourceOutput
+        RESOURCE_OUTPUT: resourceOutput,
```
Comment (rnathuji) [inline bakery/src/tasks/fetch-book-group.js:40]: It looks like this parameter is never consumed and instead we're using `.` in the script invocation.

File: bakery/src/cli/execute.js [hunk 3]
```diff
@@ -1392,7 +1344,6 @@ const yargs = require('yargs')
           .command(tasks['assemble-meta'](commandUsage))
           .command(tasks['assemble-meta-group'](commandUsage))
           .command(tasks.checksum(commandUsage))
-          .command(tasks['checksum-single'](commandUsage))
```
Comment (rnathuji) [inline bakery/src/cli/execute.js:1395]: Can we remove `checksum-single.js` as well now?

File: bakery/src/scripts/fetch_book_group.sh [hunk 4]
```diff
@@ -12,5 +12,4 @@ if [[ ! -f "${CONTENT_OUTPUT}/raw/collections/$(cat "${BOOK_INPUT}/slug").collec
 fi
 rm -rf "${CONTENT_OUTPUT}/raw/.git"
 wget "${BOOK_SLUGS_URL}" -O "${CONTENT_OUTPUT}/book-slugs.json"
-mv "${CONTENT_OUTPUT}/raw/media" "${RESOURCE_OUTPUT}/."
```
Comment (rnathuji) [inline bakery/src/scripts/fetch_book_group.sh:15]: Super minor: Should we `rmdir ${CONTENT_OUTPUT}/raw/media` after invoking `fetch-map-resources` just to clean that up?

File: bakery/src/scripts/fetch_map_resources.py [hunk 5]
```diff
@@ -1,35 +1,127 @@
...
+    """ get MIME type of file with libmagic """
+    mime_type = ''
+    try:
+        mime_type = magic.from_file(filename, mime=True)
+    finally:
+        return mime_type
+
+
+# https://stackoverflow.com/a/22058673/756056
+def get_checksums(filename):
```
Comment (rnathuji) [inline bakery/src/scripts/fetch_map_resources.py:42]: It looks like `get_checksums` and `get_mime_type` are the same (`create_json_metadata` is slightly modified) as `checksum_resource.py`. Can we import some of these from there or move to `utils.py` to share the implementation source?

**Review by rnathuji — COMMENTED** [review]
In addition to a few minor code comments, I had a doc comment: Can we update [this document image](https://github.com/openstax/output-producer-service/blob/master/bakery/docs/pipelines.png) to reflect the removal of `checksum-single`?

File: bakery/src/tasks/pdfify-single.js (hunk 1 above)
Comment (tomjw64) [reply bakery/src/tasks/pdfify-single.js:42]: And I can't believe that didn't cause an error when I ran through the steps!

File: bakery/src/scripts/checksum_resource.py [hunk 6]
```diff
@@ -1,55 +1,14 @@
 import os
 import sys
-import hashlib
-import magic
 import json
 from pathlib import Path
 from lxml import etree
+from . import utils
 
 BUF_SIZE = 8 * 1024 * 1024  # same as boto3 default chunk size. Don't modify.
```
Comment (rnathuji) [inline bakery/src/scripts/checksum_resource.py:8]: Minor cleanup: This looks like it can be removed as well from here.

**Review by rnathuji — APPROVED** [review]

**Pull request merged**